nexus-repository-r-it/src/test/java/org/sonatype/nexus/plugins/r/internal/RHostedIT.java

```diff
@@ -118,15 +120,21 @@ public void testMetadataProcessing() throws Exception
     final String agricolae131Content =
         new String(Files.readAllBytes(testData.resolveFile(PACKAGES_AGRICOLAE_131_NAME).toPath()));
 
-    //Verify PACKAGES(metadata) contain appropriate content about R package.
+    // Wait till metadata is created
+    Thread.sleep(1000);
+
+    // Verify PACKAGES(metadata) contain appropriate content about R package.
     final InputStream content = client.fetch(PACKAGES_GZ_PATH_FULL).getEntity().getContent();
     verifyTextGzipContent(is(equalTo(agricolae131Content)), content);
 
-    //Verify PACKAGES(metadata) is clean if component has been deleted
+    // Verify PACKAGES(metadata) is clean if component has been deleted
     List<Component> components = getAllComponents(repository);
     ComponentMaintenance maintenanceFacet = repository.facet(ComponentMaintenance.class);
     maintenanceFacet.deleteComponent(components.get(1).getEntityMetadata().getId());
 
+    // Wait till metadata is recreated
+    Thread.sleep(1000);
+
     final InputStream contentAfterDelete = client.fetch(PACKAGES_GZ_PATH_FULL).getEntity().getContent();
     verifyTextGzipContent(is(equalTo(agricolae121Content)), contentAfterDelete);
```

Comment:
test metadata asset exists with repo functions instead of rest API

Reply:
done in ed4f13fe223cdff2a4e750cca1cc41fbc1f95601